longevity_sla_test.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2022 ScyllaDB
+import re
+
+from longevity_test import LongevityTest
+from sdcm.sct_events import Severity
+from sdcm.sct_events.system import TestFrameworkEvent
+from test_lib.sla import Role, ServiceLevel
+
+
+class LongevitySlaTest(LongevityTest):
+    DEFAULT_USER = "cassandra"
+    DEFAULT_USER_PASSWORD = "cassandra"
+    STRESS_ROLE_NAME_TEMPLATE = 'role%d'
+    STRESS_ROLE_PASSWORD_TEMPLATE = 'rolep%d'
+    SERVICE_LEVEL_NAME_TEMPLATE = 'sl%d'
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.service_level_shares = self.params.get("service_level_shares")
+        self.roles = []
+
+    def test_custom_time(self):
+        with self.db_cluster.cql_connection_patient(node=self.db_cluster.nodes[0], user=self.DEFAULT_USER,
+                                                    password=self.DEFAULT_USER_PASSWORD) as session:
+            for shares in self.service_level_shares:
```

Comment:
This method of creation will not work if I have the same shares count twice. Maybe you should name them according to
their index in the list or maybe even arbitrarily.

Reply:
changed